salt/modules/nilrt_ip.py

```diff
@@ -0,0 +1,515 @@
+# -*- coding: utf-8 -*-
+'''
+The networking module for NI Linux Real-Time distro
+
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import logging
+import time
+
+
+log = logging.getLogger(__name__)
+
+# Import salt libs
+import salt.utils
+import salt.utils.validate.net
+import salt.exceptions
+# Import third party libs
+import pyconnman
+import dbus
+
+
+# Define the module's virtual name
+__virtualname__ = 'ip'
+
+SERVICE_PATH = '/net/connman/service/'
+_CONFIG_TRUE = ['yes', 'on', 'true', '1', True]
+
+
+def __virtual__():
+    '''
+    Confine this module to NI Linux Real-Time based distros
+    '''
+    if __grains__['os_family'] == 'NILinuxRT':
+        if _get_state() == 'offline':
+            return False, 'Connmand is not running'
+        return __virtualname__
+    return False, 'The nilrt_ip module could not be loaded: unsupported OS family'
+
+
+def _get_state():
+    try:
+        state = pyconnman.ConnManager().get_property('State')
+    except Exception as exc:
+        raise salt.exceptions.CommandExecutionError('Connman daemon error: {0}'.format(exc))
+    return state
+
+
+def _get_technologies():
+    strTech = ''
+    technologies = pyconnman.ConnManager().get_technologies()
+    for path, params in technologies:
+        strTech += '{0}\n\tName = {1}\n\tType = {2}\n\tPowered = {3}\n\tConnected = {4}\n'.format(
+            path, params['Name'], params['Type'], params['Powered'] == 1, params['Connected'] == 1)
+    return strTech
+
+
+def _add_path(service):
+    return '{0}{1}'.format(SERVICE_PATH, service)
+
+
+def _get_services():
+    '''
+    Returns a list with all connman services.
+    '''
+    serviceList = []
+    services = pyconnman.ConnManager().get_services()
+    for path, params in services:
+        serviceList.append(str(path[len(SERVICE_PATH):]))
+    return serviceList
+
+
+def _connected(service):
+    '''
+    Verify if a connman service is connected
+    '''
+    state = pyconnman.ConnService(_add_path(service)).get_property('State')
+    if state == 'online' or state == 'ready':
+        return True
+    return False
```

Comment:
We could shorten this to `return state == 'online' or state == 'ready'` which will have the same result and be a little easier to read.